haystack/nodes/prompt/invocation_layer/amazon_bedrock.py

```diff
@@ -0,0 +1,368 @@
+    """
+    Invocation layer for Amazon Bedrock models.
+    """
+
+    SUPPORTED_MODEL_PATTERNS: Dict[str, Type[BedrockModelAdapter]] = {
+        r"amazon.titan-text.*": AmazonTitanAdapter,
+        r"ai21.j2.*": AI21LabsJurassic2Adapter,
+        r"cohere.command.*": CohereCommandAdapter,
+        r"anthropic.claude.*": AnthropicClaudeAdapter,
+        r"meta.llama2.*": MetaLlama2ChatAdapter,
+    }
+
+    def __init__(
+        self,
+        model_name_or_path: str,
+        aws_access_key_id: Optional[str] = None,
+        aws_secret_access_key: Optional[str] = None,
+        aws_session_token: Optional[str] = None,
+        aws_region_name: Optional[str] = None,
+        aws_profile_name: Optional[str] = None,
+        max_length: Optional[int] = 100,
+        **kwargs,
+    ):
+        super().__init__(model_name_or_path, **kwargs)
+        self.max_length = max_length
+
+        try:
+            session = self.get_aws_session(
+                aws_access_key_id=aws_access_key_id,
+                aws_secret_access_key=aws_secret_access_key,
+                aws_session_token=aws_session_token,
+                aws_region_name=aws_region_name,
+                aws_profile_name=aws_profile_name,
+            )
+            self.client = session.client("bedrock-runtime")
+        except Exception as exception:
+            raise AmazonBedrockConfigurationError(
+                "Could not connect to Amazon Bedrock. Make sure the AWS environment is configured correctly."
+            ) from exception
+
+        model_input_kwargs = kwargs
+        # We pop the model_max_length as it is not sent to the model
+        # but used to truncate the prompt if needed
+        model_max_length = kwargs.get("model_max_length", 4096)
+
+        # Truncate prompt if prompt tokens > model_max_length-max_length
+        # (max_length is the length of the generated text)
+        # It is hard to determine which tokenizer to use for the SageMaker model
+        # so we use GPT2 tokenizer which will likely provide good token count approximation
+        self.prompt_handler = DefaultPromptHandler(
+            model_name_or_path="gpt2", model_max_length=model_max_length, max_length=self.max_length or 100
+        )
+
+        model_apapter_cls = self.get_model_adapter(model_name_or_path=model_name_or_path)
+        if not model_apapter_cls:
+            raise AmazonBedrockConfigurationError(
+                f"The model {model_name_or_path} is not supported by this invocation layer."
+            )
+        self.model_adapter = model_apapter_cls(model_kwargs=model_input_kwargs, max_length=self.max_length)
+
+    def _ensure_token_limit(self, prompt: Union[str, List[Dict[str, str]]]) -> Union[str, List[Dict[str, str]]]:
+        # the prompt for this model will be of the type str
+        if isinstance(prompt, List):
+            raise ValueError("SageMaker invocation layer doesn't support a dictionary as prompt, only a string.")
```

Comment:
The SageMaker invocation layer only supports a string as a prompt, while currently, the prompt is a dictionary.